hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/ITTestHoodieDataSource.java

```diff
@@ -2189,7 +2182,7 @@ void testReadMetaFields(HoodieTableType tableType, String queryType, int numInse
         .end();
     batchTableEnv.executeSql(readHoodieTableDDL);
 
-    List<Row> result = execSelectSql(batchTableEnv, "select * from t1", ExecMode.BATCH);
+    List<Row> result = execSelectSql(batchTableEnv, "select * from t1", ExecMode.BATCH, -1);
```

Comment:
Can we add a seperate method for it and also a new collect sink specific for function of expected num rows.